assignments/arrays.js

```diff
@@ -63,32 +63,107 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// {"id":33,"car_make":"Jeep","car_model":"Wrangler","car_year":2011},
+console.log(`Car 33 is a ` + inventory[32].car_year + ` ` + inventory[32].car_make + ` ` + inventory[32].car_model );
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+// {"id":50,"car_make":"Lincoln","car_model":"Town Car","car_year":1999}];
+let lastCar = inventory.length-1;
+console.log(`Make and model of the last car is ` + inventory[lastCar].car_make +` and `+ inventory[lastCar].car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+function sortByCarModel(){
+    let extrCar="";
+    let left = 0;
 
+    let carModels = [];
+    for (let i=0; i< inventory.length-1; i++) {
+        carModels[i] = inventory[i].car_model;
+    }
+    let right = carModels.length-1;
+
+    for (let i=0; i<carModels.length;i++) {
+
+        for (let i = left; i < right; i++) {
+            if (carModels[i].charAt(0) > carModels[i + 1].charAt(0)) {
+                extrCar = carModels[i];
+                carModels[i] = carModels[i + 1];
+                carModels[i + 1] = extrCar;
+            }
+        }
+        right--;
+        for (let j = right; j > left; j--) {
+            if (carModels[right].charAt(0) < carModels[right - 1].charAt(0)) {
+                extrCar = carModels[right];
+                carModels[right] = carModels[right - 1];
+                carModels[right - 1] = extrCar;
+                left++;
+            }
+        }
+    }
+    return carModels;
+}
+console.log(sortByCarModel());
+// OR
+function easySort(){
+    let carModels = [];
+    for (let i=0; i< inventory.length-1; i++) {
+        carModels[i] = inventory[i].car_model;
+    }
+    return "Second way to sort cars by models: " +carModels.sort();
+}
+console.log(easySort());
```

Comment:
I am really impressed by all your hard work here.  However, as you suspected, there was a simpler way.  

let carModels = []; // set up an empty array where you will put only car models in
for (let i = 0; i < inventory.length; i++) {
  carModels.push(inventory[i].car_model); // go through the inventory array, copying each car_model as you go and pushing it into the new carModels array
}
// now that we have a carModels array with only car_models in it, there is a cool sorting method you can use to sort any array.
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
console.log(carModels.sort()); // now you can use the array method .sort() to sort the array